sql/core/src/main/scala/org/apache/spark/sql/execution/python/ArrowPythonUDTFRunner.scala

```diff
@@ -34,6 +34,7 @@ class ArrowPythonUDTFRunner(
     udtf: PythonUDTF,
     evalType: Int,
     offsets: Array[Int],
+    names: Array[Option[String]],
```

Comment:
These argNames match 1:1 with the argOffsets? Could it be a bit safer to combine them into a helper case class (e.g. in case we want to add more metadata to the arguments later?). Same for other places in this PR.